azurerm/resource_arm_dns_zone.go

```diff
@@ -45,6 +45,29 @@ func resourceArmDnsZone() *schema.Resource {
 				Set:      schema.HashString,
 			},
 
+			"zone_type": {
+				Type:     schema.TypeString,
+				Default:  nil,
+				Optional: true,
+				Computed: true,
```

Comment:
can we:

1) update the Default to be `string(cdn.Public)` - this'll ensure compatibility with existing resources
2) remove the `Computed: true` (since we're specifying a Default value)
3) add a ValidateFunc to ensure only Public and Private can be specified, this can be done via:
```
ValidateFunc: validation.StringInSlice([]string{
  string(cdn.Private),
  string(cdn.Public),
}, true)
```
4) add a `DiffSuppressFunc` here to handle case insensitivity (since this is an Enum) - this can be done via:

```
DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```

Reply:
done.